### ISSUE TITLE
[KubeRay, VM launcher] Provide clear guidelines for whether to have multi-tenant, long-running clusters for use cases other than Ray Serve 

### ISSUE THREAD

**Issue by scottsun94** [ray-project/ray#35250]
### Description

Users want clearer guidelines around whether to set up long-running, multi-tenant Ray Clusters. We should make it clear that we don't recommend long-running clusters for non-ray-serve use cases.
https://github.com/ray-project/ray/issues/35202#issuecomment-1542932603

cc: @gvspraveen @architkulkarni @kaisu

**Comment by scottsun94** [comment]
https://docs.ray.io/en/latest/cluster/faq.html#do-ray-clusters-support-multi-tenancy
We do have doc about multi-tenancy which is not recommended for the production use case.

Whether people should set up long-running clusters is still not clear

**Comment by gvspraveen** [comment]
Agree. Our [documentation](https://docs.ray.io/en/latest/cluster/faq.html#do-ray-clusters-support-multi-tenancy) also says `On the other hand, you can run the same job multiple times using the same cluster to save the cluster startup time`. But it is not clear if this is directly recommending long running clusters.


I heard of few use cases in [Kuberay](https://github.com/ray-project/kuberay) where we recommending using RayCluster resources in kuberay to setup long running cluster and submit multiple jobs.
